tests/cloner_test.go

```diff
@@ -521,6 +521,107 @@ var _ = Describe("Block PV Cloner Test", func() {
+		config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		config.Spec.PodResourceRequirements = orgConfig
+		_, err = f.CdiClient.CdiV1alpha1().CDIConfigs().Update(config)
+		Expect(err).ToNot(HaveOccurred())
+		if sourcePvc != nil {
+			By("[AfterEach] Clean up source PVC")
+			err := f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if targetPvc != nil {
+			By("[AfterEach] Clean up target PVC")
+			err := f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Should create clone in namespace with quota", func() {
+		f.CreateQuotaInNs(int64(1), int64(1024*1024*1024), int64(2), int64(2*1024*1024*1024))
+		smartApplicable := f.IsSnapshotStorageClassAvailable()
+		sc, err := f.K8sClient.StorageV1().StorageClasses().Get(f.SnapshotSCName, metav1.GetOptions{})
+		if err == nil {
+			value, ok := sc.Annotations["storageclass.kubernetes.io/is-default-class"]
+			if smartApplicable && ok && strings.Compare(value, "true") == 0 {
+				Skip("Cannot test host assisted cloning for within namespace when all pvcs are smart clone capable.")
+			}
+		}
+
+		pvcDef := utils.NewPVCDefinition(sourcePVCName, "1G", nil, nil)
+		pvcDef.Namespace = f.Namespace.Name
+		sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile)
+		doFileBasedCloneTest(f, pvcDef, f.Namespace)
+	})
+
+	It("Should fail to clone in namespace with quota when pods have higher requirements", func() {
+		f.UpdateCdiConfigResourceLimits(int64(2), int64(1024*1024*1024), int64(2), int64(1*1024*1024*1024))
+		f.CreateQuotaInNs(int64(1), int64(1024*1024*1024), int64(2), int64(2*1024*1024*1024))
+		smartApplicable := f.IsSnapshotStorageClassAvailable()
+		sc, err := f.K8sClient.StorageV1().StorageClasses().Get(f.SnapshotSCName, metav1.GetOptions{})
+		if err == nil {
+			value, ok := sc.Annotations["storageclass.kubernetes.io/is-default-class"]
+			if smartApplicable && ok && strings.Compare(value, "true") == 0 {
+				Skip("Cannot test host assisted cloning for within namespace when all pvcs are smart clone capable.")
+			}
+		}
+
+		pvcDef := utils.NewPVCDefinition(sourcePVCName, "1G", nil, nil)
+		pvcDef.Namespace = f.Namespace.Name
+		sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile)
+		// Create targetPvc in new NS.
+		targetDV := utils.NewCloningDataVolume("target-dv", "1G", pvcDef)
+		dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDV)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		By("Verify Quota was exceeded in logs")
+		matchString := fmt.Sprintf("upload pod API create errored: pods \"cdi-upload-target-dv\" is forbidden: exceeded quota: test-quota, requested: requests.cpu=2, used: requests.cpu=0, limited: requests.cpu=1")
```

Comment:
It might bite us in the future if we're hard-coding the full error string, if they change it even slightly

Reply:
So its the test that is creating the name that we are checking for. I like to have a different mechanism (in this case hard coding) than what the actual code is doing for checking value against. That way if there is an issue with the mechanism that generates the value, the test doesn't pass because both result and match used the same mechanism to generate the value.